pkg/util/validation/limits.go

```diff
@@ -787,3 +790,59 @@ func MaxDurationPerTenant(tenantIDs []string, f func(string) time.Duration) time
 	}
 	return result
 }
+
+type userLimitsResponse struct {
```

Comment:
WDYT about exposing `CompactorBlocksRetentionPeriod` as well?

With this information a tenant will be able to know for how long its data will be keep. I would say this should the retention period in seconds (int64). We have the case of unlimited retention `CompactorBlocksRetentionPeriod=0`, we could either return `0` as it is and point to the documentation where it is explained that `0` means unlimited or we can return `math.MaxInt64`. The former seems better.

Example of use of this information:
* In Grafafa with a `JSON API` data source type, I can GET this information and render it as a single stat while choosing the unit `Time / duration (s)`

Reply:
I think exposing `CompactorBlocksRetentionPeriod` is fine, with 0 meaning "unlimited". It's consistent with config option.

Would it make sense to move this handler and struct to a separate file in this package?